scipy/signal/signaltools.py

```diff
@@ -873,10 +873,52 @@ def lfilter(b, a, x, axis=-1, zi=None):
     """
     if isscalar(a):
         a = [a]
-    if zi is None:
-        return sigtools._linear_filter(b, a, x, axis)
+    if a[0] == 1 and len(a) == 1:
+        # This path only supports types fdgFDGO to mirror _linear_filter below.
+        # Any of b, a, x, or zi can set the dtype, but there is no default 
+        # casting of other types; instead a NotImplementedError is raised.
+        b = np.asarray(b)
+        a = np.asarray(a)
+        if b.ndim != 1 and a.ndim != 1:
+            raise ValueError('object of too small depth for desired array')
+        x = np.asarray(x)
+        inputs = [b, a, x]
+        if zi is not None:
+            # _linear_filter does not broadcast zi
+            zi = np.asarray(zi)
+            if zi.ndim != x.ndim:
+                raise ValueError('object of too small depth for desired array')
+            expected_shape = list(x.shape)
+            expected_shape[axis] = max(a.shape[0], b.shape[0]) - 1
```

Comment:
``a.shape[0]`` is always 1, so never larger then ``b.shape[0]`` --> can be simplified.